docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler

```diff
@@ -0,0 +1,502 @@
+# Onboarding a Spring Boot based REST API Service
+
+This guide is part of a series of guides to onboard a REST API service with the Zowe API Mediation Layer.
+As an API developer, you can onboard your REST API service built with the Spring Boot framework with the Zowe API Mediation Layer.
+
+**Note:** Before version 1.12, the API ML provided an integration enabler based on Spring Cloud Netflix components. From version 1.12 and later, the enabler has been replaced with new implimentation based on the Plain Java Enabler (PJE) that is not backwards compatiable with the previous enabler versions.
+
+
+**Tip:** For more information about how to utilize another onboarding method, see:
+
+  * [Onboard a REST API service with the Plain Java Enabler (PJE)](api-mediation-onboard-an-existing-java-rest-api-service_plain-java-enabler.md)
+  * [Onboard a REST service directly calling eureka with xml configuration](api-mediation-onboard-rest-service-direct-eureka-call.md)
+  * [Onboard an existing REST API service without code changes](api-mediation-onboard-an-existing-rest-api-service-without-code-changes.md)
+  * [Java REST APIs service without Spring Boot](api-mediation-onboard-an-existing-java-rest-api-service-without-spring-boot-with-zowe-api-mediation-layer.md)
+
+## Outline of onboarding a REST service using Spring Boot
+
+The following steps outline the overall process to onboard a REST service with the API ML using a Spring Boot enabler. Each step is described in further detail in this article.
+
+1. [Selecting a Spring Boot Enabler](#selecting-an-spring-boot-enabler)
+
+2. [Configuring your project](#configuring-your-project)
+
+    * [Gradle guide](#gradle-guide)
+    * [Maven guide](#maven-guide)
+
+3. [Configuring your Spring Boot based service to onboard with API ML](#configuring-your-spring-boot-based-service-to-onboard-with-api-ml)
+
+4. [Registering and unregistering your service with API ML](#registering-and-unregistering-your-service-with-api-ml)
+
+5. [Adding API documentation](#adding-api-documentation)
+
+6. (Optional) [Validating your API service discoverability](#validating-the-discoverability-of-your-api-service-by-the-discovery-service)
+
+
+## Selecting a Spring Boot Enabler
+
+Add to your project build configuration a dependency on Spring Enabler version corresponding to the Spring Boot version, that you use for the whole project:
+- onboarding-enabler-spring-v1
+- onboarding-enabler-spring-v2
+
+**Note:** The process of onboarding an API service is the same for both Spring Boot enabler versions.
+
+## Configuring your project
+
+Use either _Gradle_ or _Maven_ build automation systems to manage your project builds.
+
+**Note:** You can download the selected enabler artifact from the Giza Artifactory or  if you decide to build the API ML from source, you are required to publish the enabler artifact to your Artifactory.
+ Publish the enabler artifact by using the _Gradle_ tasks provided in the source code.
```

Comment:
We can discuss around these points;
Do we need to mention which gradle publish artifact ?

Reply:
The idea behind is, that this is an open source, so if the user decides to clone/fork the source code and make changes, they can, but then they need to publish to a artifactory to which they have write access.